### PR TITLE
APP-3000: Simplified configuration when the configuration of api client is the same

### PR DESCRIPTION
**Pull request by symphony-hong**
- 4 properties added to BdkConfig to specify the global configurations for host,
port, scheme and context. These properties can be used if they are not configured
in the api client configuration.

- Version of configuration file is also removed because it seems useless.

- Unittest updated

- Documentation updated

### CODE REVIEW

File: symphony-bdk-core/src/test/resources/config/config_client_global.yaml [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+
+pod:
+  port: 8443
+
+agent:
+  host: devx1.symphony.com
+  port: 443
+
+keyManager:
+  host: devx1.symphony.com
```
Comment (symphony-thibault) [inline symphony-bdk-core/src/test/resources/config/config_client_global.yaml:13]: You add a test case where the `context` is different

File: symphony-bdk-examples/bdk-core-examples/src/main/resources/rsa-private-fxBot.pem [hunk 2]
```diff
@@ -0,0 +1,46 @@
+-----BEGIN RSA PRIVATE KEY-----
```
Comment (symphony-thibault) [inline symphony-bdk-examples/bdk-core-examples/src/main/resources/rsa-private-fxBot.pem:1]: Rename this file to `sample-privatekey.pem`

Comment (symphony-hong) [reply symphony-bdk-examples/bdk-core-examples/src/main/resources/rsa-private-fxBot.pem:1]: Oops! This file was commit by a mistake.

File: symphony-bdk-core/src/test/resources/config/config_client_global.yaml (hunk 1 above)
Comment (symphony-hong) [reply symphony-bdk-core/src/test/resources/config/config_client_global.yaml:13]: Added 👍

**Review by symphony-thibault — COMMENTED** [review]
Could you also modify one of the example in order to reflect this enhancement?

File: symphony-bdk-core/src/test/java/com/symphony/bdk/core/config/BdkConfigLoaderTest.java [hunk 3]
```diff
@@ -84,4 +85,16 @@ public void loadFromClasspathNotFoundTest() throws BdkConfigException {
         });
         assertEquals(exception.getMessage(), "Config file is not found");
     }
+
+    @Test
+    public void loadClientGlobalConfig() throws BdkConfigException {
```
Comment (symphony-youri) [inline symphony-bdk-core/src/test/java/com/symphony/bdk/core/config/BdkConfigLoaderTest.java:90]: with junit 5 classes and methods dont need the public modifier

File: symphony-bdk-core/src/test/java/com/symphony/bdk/core/config/BdkConfigLoaderTest.java [hunk 4]
```diff
@@ -84,4 +85,16 @@ public void loadFromClasspathNotFoundTest() throws BdkConfigException {
...
+        BdkConfig config = BdkConfigLoader.loadFromClasspath("/config/config_client_global.yaml");
+        assertEquals(config.getPod().getHost(), "devx1.symphony.com");
+        assertEquals(config.getPod().getPort(), 8443);
+        assertEquals(config.getPod().getContext(), "context");
+        assertEquals(config.getScheme(), "https");
+        assertEquals(config.getAgent().getPort(), 443);
+        assertEquals(config.getAgent().getFormattedContext(), "/context");
+        assertEquals(config.getKeyManager().getFormattedContext(), "/diff-context");
```
Comment (symphony-youri) [inline symphony-bdk-core/src/test/java/com/symphony/bdk/core/config/BdkConfigLoaderTest.java:98]: do we test all the possible cases here? (like if the pod's host can be overriden?)
(ok I'm being picky but maybe it is worth it for the SDK)

**Review by symphony-youri — APPROVED** [review]

File: symphony-bdk-core/src/test/java/com/symphony/bdk/core/config/BdkConfigLoaderTest.java (hunk 4 above)
Comment (symphony-hong) [reply symphony-bdk-core/src/test/java/com/symphony/bdk/core/config/BdkConfigLoaderTest.java:98]: I will add more test case here.

File: docs/configuration.md [hunk 5]
```diff
@@ -72,6 +72,58 @@ The configuration file after being read will be represented by `BdkConfig` class
...
+
+pod:
+  host: devx1.symphony.com
+  port: 443
+
+agent:
+  context: agent
+
+keyManager:
+  host: sym-devx1-dev-chat-glb-1-ause1-all.symphony.com
```
Comment (symphony-thibault) [inline docs/configuration.md:90]: Not sensitive information, but remember that this is an open-source project. Could you put something less "technical" here?

**Review by symphony-thibault — APPROVED** [review]
LGTM 👍

**Pull request merged**